ramips: add support for Zyxel LTE7490-M904

es86de opened the pull request:
The Zyxel LTE7490-M904 is an 802.3at PoE powered LTE outdoor (IP68) CPE with integrated directional antennas.

### Specifications
 - SoC: MediaTek MT7621AT
 - RAM: 256 MB
 - Flash: 128 MB MB NAND (MX30LF1G18AC)
 - WiFi: MediaTek MT7603E 802.11b/g/n
 - Switch: 1 LAN port (1 Gbps)
 - LTE/3G/2G: Quectel EG18-EA LTE-A Cat. 18 connected by USB3 to SoC
 - SIM: 1 micro-SIM slots under transparent cover
 - Buttons: Reset, WLAN under same cover
 - LEDs: Multicolour green/red/amber under same cover (visible)
 - Power: 802.3at PoE via LAN port

The device is built as an outdoor ethernet to LTE bridge or router. The Wifi interface is intended for installation and/or temporary management purposes only.

### UART Serial
57600N1, located on populated 5 pin header J5:
```
[o] GND
[ ] key - no pin
[o] RX
[o] TX
[o] 3.3V Vcc
```

Remove the SIM/button/LED cover and 12 screws holding the back plate and antenna cover together. Be careful with the cables.

### Installation from OEM web GUI
- Log in as "admin" on OEM web GUI
- Upload OpenWrt initramfs-recovery.bin image on the Maintenance -> Firmware page
- Wait for OpenWrt to boot and ssh to root@192.168.1.1
- Sysupgrade to the OpenWrt sysupgrade image and reboot

For more details about flashing see:
https://github.com/openwrt/openwrt/commit/2449a632084b29632605e5a79ce5d73028eb15dd (ramips: mt7621: Add support for ZyXEL NR7101, 2021-04-19)

Main porting work done by @ErnyTech:
https://github.com/openwrt/openwrt/commit/bf1c12f68b2678647a8e7437cb59907e4677122e (ramips: add support for ZyXEL LTE7490-M904, 2023-12-20)

Neustradamus commented:
@es86de: Excellent!

hauke commented on target/linux/ramips/dts/mt7621_zyxel_lte7490-m904.dts:
```diff
@@ -0,0 +1,181 @@
+		compatible = "gpio-leds";
+
+		system_yellow {
+			label = "yellow:system";
```
Please use the new LED function definition:
```
			function = LED_FUNCTION_POWER;
			color = <LED_COLOR_ID_YELLOW>;
```
same for the others

es86de replied:
Thanks, done!

es86de commented:
I just made some minor changes to the commit message. Should be ready to merge.

es86de commented:
> @bmork, @namiltd, @Noltari, @svanheule : Do you like to review? Thanks!

@mkrle, @4IceG: Maybe you? Thank you!

hauke commented:
Please rebase to fix the conflict.

es86de committed:
ramips: add support for Zyxel LTE7490-M904

The Zyxel LTE7490-M904 is an 802.3at PoE powered LTE outdoor (IP68) CPE
with integrated directional antennas.

Specifications:

- SoC: MediaTek MT7621AT
- RAM: 256 MB
- Flash: 128 MB MB NAND (MX30LF1G18AC)
- WiFi: MediaTek MT7603E 802.11b/g/n
- Switch: 1 LAN port (1 Gbps)
- LTE/3G/2G: Quectel EG18-EA LTE-A Cat. 18 connected by USB3 to SoC
- SIM: 1 micro-SIM slots under transparent cover
- Buttons: Reset, WLAN under same cover
- LEDs: Multicolour green/red/amber under same cover (visible)
- Power: 802.3at PoE via LAN port

The device is built as an outdoor ethernet to LTE bridge or router.
The wifi interface is intended for installation and/or temporary
management purposes only.

UART Serial:

57600N1, located on populated 5 pin header J5:

 [o] GND
 [ ] key - no pin
 [o] RX
 [o] TX
 [o] 3.3V Vcc

Remove the SIM/button/LED cover and 12 screws holding the back plate
and antenna cover together. Be careful with the cables.

Installation from OEM web GUI:

- Log in as "admin" on OEM web GUI
- Upload OpenWrt initramfs-recovery.bin image on the
  Maintenance -> Firmware page
- Wait for OpenWrt to boot and ssh to root@192.168.1.1
- Sysupgrade to the OpenWrt sysupgrade image and reboot

For more details about flashing see:
2449a63208 (ramips: mt7621: Add support for ZyXEL NR7101, 2021-04-19)

Main porting work done by Ernesto Castellotti <ernesto@castellotti.net>:
bf1c12f68b (ramips: add support for ZyXEL LTE7490-M904, 2023-12-20)

Signed-off-by: Eric Schäfer <eric@es86.de>
Link: https://github.com/openwrt/openwrt/pull/17485
Signed-off-by: Hauke Mehrtens <hauke@hauke-m.de>

es86de commented:
> Please rebase to fix the conflict.

Done.

Neustradamus commented:
Thanks to @es86de, good job and @hauke for merging!
